core/src/Kokkos_OpenMPTargetSpace.hpp

```diff
@@ -217,41 +217,47 @@ template <class ExecutionSpace>
 struct DeepCopy<Kokkos::Experimental::OpenMPTargetSpace,
                 Kokkos::Experimental::OpenMPTargetSpace, ExecutionSpace> {
   DeepCopy(void* dst, const void* src, size_t n) {
-    omp_target_memcpy(dst, const_cast<void*>(src), n, 0, 0,
-                      omp_get_default_device(), omp_get_default_device());
+    if (n)
```

Comment:
Readability
```suggestion
    if (n > 0)
```

Reply:
I'd like you to add a comment in the code where you elaborate on why that call to `omp_target_memcpy` is protected.  The specification does not explicitly calls out whether it is legal to pass null pointers with a zero length (see https://www.openmp.org/spec-html/5.1/openmpsu172.html).  Did you check whether this was the issue?

Also, any good reason to ignore the return value here?

Reply:
Yes the specification does not say anything specific about passing zero length to `omp_target_memcpy`. I guess the Release build just protects against unnecessary copies. 
I added a return value to the memcpy calls and am checking for its success.